### ISSUE TITLE
How to load dataset having 1000's of numpy files

### ISSUE THREAD

**Issue by ultrai** [timeseriesAI/tsai#179]
Dear TSAI team
Good job by TSAI team for bringing NLP methodologies to signal processing tasks.
This helped my research team to avoid reinventing the wheel.
The notebooks are self-explanatory. I have gone through ipynbs for loading data completely onto ram or load a single huge numpy file on harddisk. But in general scenarios we have 1000s of samples which need to be loaded recursively for each epoch.

The working  data loader for Pytorch is

```
class Dataset(torch.utils.data.Dataset):
  def __init__(self, list_IDs, labels):
        self.labels = labels
        self.list_IDs = list_IDs

  def __len__(self):
        return len(self.list_IDs)

  def __getitem__(self, index):
        # Select sample
        **ID = self.list_IDs[index]
        X = np.load(df_train['Location'].loc[ID])
        y = self.labels[ID]**
        return X, y   

training_set = Dataset(partition['train'], labels)
training_generator = torch.utils.data.DataLoader(training_set, **params)

for x,y in training_generator:
    print(x)

```

Could you provide a pointer for converting this above classical dataloader to TSAI model acceptable dataloader

**Comment by TKassis** [comment]
Please see #126

**Comment by oguiza** [comment]
Hi @ultrai ,  (cc:@TKassis)

There are 2 classes (TSMetaDataset and TSMetaDatasets) that were built with this idea in mind. Sometimes you have lots of input files instead of a single array. 

I've put together a [notebook](https://colab.research.google.com/gist/oguiza/51700e4b219ddcf19b77613fd8b14d41/11_how_to_use_multiple_input_files_on_disk.ipynb) to demonstrate this functionality. It'd be good it you would test it with a large amount of files. I'd love to get your feedback.

If/ when this works well (I expect some issues), I'll expand it into a new tutorial book.

**Comment by ultrai** [comment]
@TKassis 
It worked as a charm without much change thanks

@oguiza 
Your solution worked too, Thanks. The Label should be wrapped as an array if each .npy array has only one sample. However its pretty straightforward and simple.
